src/viam/app/data/client.py

```diff
@@ -290,10 +320,41 @@ async def bounding_box_labels_by_filter(self, filter: Optional[Filter] = None) -
         response: BoundingBoxLabelsByFilterResponse = await self._data_client.BoundingBoxLabelsByFilter(request, metadata=self._metadata)
         return response.labels
 
-    # TODO(RSDK-3637): Implement
-    async def data_capture_upload(self, metadata: UploadMetadata, sensor_contents: Optional[List[SensorData]] = None) -> None:
-        raise NotImplementedError()
+    async def data_capture_upload(self, metadata: UploadMetadata , sensor_contents: List[SensorData]) -> None:
+        """Upload binary or tabular sensor data.
+
+        Sync binary or tabular data collected on a robot through a specific component (i.e. a motor) along with the relevant metadata with
+        app.viam.com.
+
+        Args:
+            metadata (viam.app.proto.datasync.UploadMetaData): Metadata relating the data collected with the corresponding robot part. Must
+                specify a part ID, component type, component name, method name, and data type.
+            sensor_contents (List[viam.app.proto.datasync.SensorData]): A list of the data to be uploaded. If the metadata specifies a
+                binary data type, the "binary" field of sensor_contents will be uploaded. Alternatively, the "struct" field will be uploaded
+                if the data type is tabular. Length of sensor_contents must be 1 if uploading a binary data capture. Otherwise, must simply
+                be nonempty.
+
+        Raises:
+            GRPCError: If a required data field is not defined.
+        """
+        request = DataCaptureUploadRequest(metadata=metadata, sensor_contents=sensor_contents)
+        _: DataCaptureUploadResponse = await self._data_sync_client.DataCaptureUpload(request, metadata=self._metadata)
 
-    # TODO(RSDK-3637): Implement
     async def file_upload(self, metadata: UploadMetadata, file_contents: Optional[FileData]) -> None:
```

Comment:
One other thought: I think maintaining this version of `file_upload` makes sense, but also it would be great if we had a method that instead took as an argument a path to a file, and then we could convert its contents into bytes under the hood and infer the file name and extension and just upload that. Seems like a much more pleasant user experience.

Reply:
I like this. Do you want to overload the function? Also quick question: by "this version" do you mean the new one where we create the metadata and filedata objects ourselves under the hood? Or the old version where the client passes them as already existing arguments.